starfish/test/image/filter/test_api_contract.py

```diff
@@ -0,0 +1,106 @@
+"""
+This module stores a contract for new Filter algorithms.
+
+Contract
+--------
+
+- constructor accepts 3d and 2d data, where default is 2d
+- values emitted by a filter are floats between 0 and 1 (inclusive)
+- exposes a `run`() method
+- run accepts an in-place parameter which defaults to True
+- run always returns an ImageStack (if in-place, returns a reference to the modified input data)
+- run accepts an `n_processes` parameter which determines
+- run accepts a `verbose` parameter, which triggers tqdm to print progress
+
+To add a new filter, simply add default
+parameters for the constructor (omit is_volume), and it will be tested against the contract by
+way of registration to the FilterAlgorithmBase
+"""
+
+from typing import Dict
+
+import numpy as np
+import pytest
+
+from starfish import ImageStack
+from starfish.image import Filter
+
+# import pdb; pdb.set_trace()
+methods: Dict = Filter._algorithm_to_class_map()
+
+# store some default parameters for testing purposes. New algorithms will have to add parameters to
+# this in order to not fail tests
+default_filtering_parameters = {
+    'Bandpass': dict(lshort=1, llong=3, threshold=0.01),
```

Comment:
Does it make sense to put these defaults in the algorithms classes?

Reply:
Potentially as a private constant `_TESTING_DEFAULTS`. Definitely not exposed to users. I like the former, what do you think?